plugins/inputs/modbus/modbus.go

```diff
@@ -18,51 +17,46 @@ import (
 
 // Modbus holds all data relevant to the plugin
 type Modbus struct {
-	Name             string           `toml:"name"`
-	Controller       string           `toml:"controller"`
-	TransmissionMode string           `toml:"transmission_mode"`
-	BaudRate         int              `toml:"baud_rate"`
-	DataBits         int              `toml:"data_bits"`
-	Parity           string           `toml:"parity"`
-	StopBits         int              `toml:"stop_bits"`
-	SlaveID          int              `toml:"slave_id"`
-	Timeout          config.Duration  `toml:"timeout"`
-	Retries          int              `toml:"busy_retries"`
-	RetriesWaitTime  config.Duration  `toml:"busy_retries_wait"`
-	DiscreteInputs   []fieldContainer `toml:"discrete_inputs"`
-	Coils            []fieldContainer `toml:"coils"`
-	HoldingRegisters []fieldContainer `toml:"holding_registers"`
-	InputRegisters   []fieldContainer `toml:"input_registers"`
-	Log              telegraf.Logger  `toml:"-"`
-	registers        []register
-	isConnected      bool
-	tcpHandler       *mb.TCPClientHandler
-	rtuHandler       *mb.RTUClientHandler
-	asciiHandler     *mb.ASCIIClientHandler
-	client           mb.Client
-}
-
-type register struct {
-	Type           string
-	RegistersRange []registerRange
-	Fields         []fieldContainer
-}
-
-type fieldContainer struct {
-	Measurement string   `toml:"measurement"`
-	Name        string   `toml:"name"`
-	ByteOrder   string   `toml:"byte_order"`
-	DataType    string   `toml:"data_type"`
-	Scale       float64  `toml:"scale"`
-	Address     []uint16 `toml:"address"`
+	Name             string          `toml:"name"`
+	Controller       string          `toml:"controller"`
+	TransmissionMode string          `toml:"transmission_mode"`
+	BaudRate         int             `toml:"baud_rate"`
+	DataBits         int             `toml:"data_bits"`
+	Parity           string          `toml:"parity"`
+	StopBits         int             `toml:"stop_bits"`
+	Timeout          config.Duration `toml:"timeout"`
+	Retries          int             `toml:"busy_retries"`
+	RetriesWaitTime  config.Duration `toml:"busy_retries_wait"`
+	Log              telegraf.Logger `toml:"-"`
+	// Register configuration
+	ConfigurationOriginal
+	// Connection handling
+	client      mb.Client
+	handler     mb.ClientHandler
+	isConnected bool
+	// Request handling
+	requests map[byte]requestSet
+}
+
+type fieldConverterFunc func(bytes []byte) interface{}
+
+type requestSet struct {
+	coil     []request
+	discrete []request
+	holding  []request
+	input    []request
+}
+
+type field struct {
```

Comment:
is this still used?

Reply:
It's a but unfortunate, but yes, the `requestSet` has a list of `request` instances and the `request` (see _request.go_) has a list of `field`s associated with / filled by that request. I thought moving out the request would be good, but if you like it more to have it in here it's only the type-definition and one function.